### ISSUE TITLE
Crash - Reliable, sometimes persistent crashing the browser via JavaScript

### ISSUE THREAD

**Issue by mferrera** [mozilla-mobile/firefox-ios#8744]
## Steps to reproduce
Open a page with the following JavaScript when the `charset` is set to utf-8.
```
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
...
<script type="text/javascript">
    var bad_char = "𝍖";
    document.title = bad_char.charAt(0);
</script>
```
Ex: https://mferrera.keybase.pub/firefox.html

### Expected behavior
`document.title` is set without crashing.

### Actual behavior
The browser crashes.

### Device & build information
* Device: 
  * iPhone 7 / iOS 14.6
  * iPad Pro / iOS 14.4
  * iPad / iOS 14.6
 
* Build version: 
  * 34.1 / 5186
  * 34.0 / 4920
  * 34.1 / 5186 
  * (respectively)
 
### Notes
At one point while testing this I managed to enter a state where I could not open the app whatsoever: it crashed immediately without giving the "hey we crashed, want to reopen your tabs?" prompt. I had to reinstall it.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-1982)

**Comment by kcome** [comment]
After testing, this issue is caused by the same reason that causes [this one](https://github.com/mozilla-mobile/firefox-ios/issues/8723).

A pull-request was created. When it's merge this issue could be closed. 

cc @SimonBasca

**Comment by SimonBasca** [comment]
This issue should have been closed when the PR was merged. Closing now as verified fix on v98.0 (8233) and it's already fixed in release 97.0.
